src/net/gateway.ts

```diff
@@ -143,6 +153,22 @@ export class Gateway {
         return this.web3.isReady && this.dvote.isReady
     }
 
+    public getArchiveUri(): string {
```

Comment:
- This could be a getter and be used like `myGw.archiveUri` instead of `myGw.getArchiveUri()`
- The internal variable might have to be called `_archiveIpndId`